samples/packages/awscdk-typescript/src/package-config.ts

```diff
@@ -127,46 +131,47 @@ export class PackageComponentKeySoftwareConfig implements PackageComponentConfig
     }
   }
 
-  // getAwsKeyUsage(): KeyUsage {
-  //   return PackageComponentKeySoftwareConfig.awsKeyUsage(this.usage);
-  // }
 }
 
 export class PackageComponentMessageQueueConfig implements PackageComponentConfig {
+  static TYPING = (manifest as PackageManifest).constructs.components.types[ComponentTyping.MESSAGE_QUEUE];
 
   constructor(
     public readonly name: string,
     public readonly fifo?: boolean,
     public readonly dlq?: number,
-  ) { }
+  ) {}
 
   public static has(config: cdf.TraitConfig<PackageComponentConfig>): boolean {
-    return config.type == 'message' && config.subtype == 'queue';
+    return config.type == this.TYPING.type && config.subtype == this.TYPING.subtype;
   }
 }
 
 export class PackageComponentMessageTopicConfig implements PackageComponentConfig {
+  static TYPING = (manifest as PackageManifest).constructs.components.types[ComponentTyping.MESSAGE_TOPIC];
 
   constructor(
     public readonly name: string,
     public readonly fifo?: boolean,
-  ) { }
+  ) {}
 
   public static has(config: cdf.TraitConfig<PackageComponentConfig>): boolean {
-    return config.type == 'message' && config.subtype == 'topic';
+    return config.type == this.TYPING.type && config.subtype == this.TYPING.subtype;
   }
 }
 
 
 export class PackageComponentStorageBucketConfig implements PackageComponentConfig {
+  static TYPING = (manifest as PackageManifest).constructs.components.types[ComponentTyping.STORE_OBJECT];
+  
   constructor(
     public readonly name: string,
     public readonly bucketName: string,
     public readonly blockPublicAccess?: ComponentBlockPublicAccess,
-  ) { }
+  ) {}
 
   public static has(config: cdf.TraitConfig<PackageComponentConfig>): boolean {
-    return config.type == 'storage' && config.subtype == 'bucket';
+    return config.type == this.TYPING.type && config.subtype == this.TYPING.subtype;
```

Comment:
In particular for this component type (object storage): *I've changed both type and subtype to be generic, to `store` and `object` consistent with the google sheets that contain all the services being researched. * 
It's probably okay here to call it *bucket* given the AWS context, but let's set an example to use generic typing to indicate that there's no need to be tied to underlying building blocks.